### ISSUE TITLE
How to add files to the rootfs without interaction?

### ISSUE THREAD

**Issue by T-vK** [livecd-tools/livecd-tools#209]
I have a script that needs to automatically modify a Fedora ISO without user interaction, adding a few files to the rootfs.
The only way to do this appeared to be to unpack the squashfs image, mount the rootfs image, add my files through the mountpoint, unmount the rootfs image again, make a new squashfs image containing the modified rootfs image, and then finally run `editliveos` with the `--rootfsimg /path/to/the/modified/squashfs.img` flag.

But first of all that takes waay too long because `editliveos` itself again unpacks the squashfs and rebuilds it.
And secondly my changes don't make it to the final ISO, suggesting that the `--rootfsimg` parameter didn't do anyhting.

If I manually add my modified squashfs.img to the ISO using `xoriso`, it works fine.

But then I still have to use `editliveos` afterwards because I need it for the `--extra-kernel-args`.

Isn't there a simple way to do all of that with one call to `editliveos`? 

I've been looking the `--script` option, but a script running in a chroot of the rootfs isn't gonna help me copy files from the host into it...

**Comment by FGrose** [comment]
There are bindmounts in the chroot to the host filetree.  One is `/dev/shm`.
For example, if you had `--script /tmp/cp.sh`, in your arguments, and it bore the following code:
```sh
#!/bin/bash
cp /dev/shm/somefile /somedirectory/somefile
```
then `/somedirectory` in the rootfs would hold `somefile`.

**Comment by T-vK** [comment]
Thank you, I will give that a try!

**Comment by T-vK** [comment]
I'm not having any luck with this. It can't seem to find my files:

```
cp: cannot stat '/dev/shm/tmp/get-mbpt.sh': No such file or directory
cp: cannot stat '/dev/shm/tmp/mbpt.desktop': No such file or directory
cp: cannot stat '/dev/shm/tmp/mbpt.desktop': No such file or directory
```

``` bash
#!/usr/bin/env bash

mkdir -p /etc/skel/.config/autostart/
cp /dev/shm/tmp/get-mbpt.sh /etc/skel/
cp /dev/shm/tmp/mbpt.desktop /etc/skel/.config/autostart/
cp /dev/shm/tmp/mbpt.desktop /usr/share/applications/
```

**Comment by T-vK** [comment]
I just added an ls to the script and `/dev/shm` is basically empty:

```
$ ls -la /dev/shm/
total 4
drwxrwxrwt. 2 root root   60 Aug 17 09:50 .
drwxr-xr-x. 4 root root 4096 Aug 17 09:51 ..
-rw-rw----. 1 1000 kvm     0 Aug 16 11:18 looking-glass
```

This is definitely not the root directory of my host system.

**Comment by FGrose** [comment]
> This is definitely not the root directory of my host system.

True, the bindmount is from the host's `/dev/shm` to the chroot's `/dev/shm`.
So, your source files should be placed in the host `/dev/shm` directory.

**Comment by T-vK** [comment]
I see. Not ideal, but I guess it'll do the trick. Thanks again. :)

**Comment by FGrose** [comment]
If you want to change lines 1400-1407 of editliveos to the following:
```python
bindmounts = [('/sys', None), ('/proc', None), ('/dev/pts', None),
              ('/dev/shm', None), ('/run', None),
              ('/etc/resolv.conf', None),
              (cachesrc, '/var/cache/dnf'),
              (cachesrc, '/var/cache/yum'),
              (self._LiveImageCreatorBase__isodir, '/mnt/live'),
              (self.srcmntdir, '/run/initramfs/live'),
              ('/', '/mnt/hostroot')]
 ```
(adding `,
              ('/', '/mnt/hostroot')` to the bindmounts list),
You should be able to use `/mnt/hostroot/a_host_directory/somefile` as your source in the script.

**Comment by FGrose** [comment]
PR #214 includes the `/run/hostroot` bindmount.

**Comment by T-vK** [comment]
It's been a long time, but I guess I got it to work. Check the build_method_1 function in this script: https://github.com/T-vK/MobilePassThrough/blob/master/scripts/main/generate-live-iso.sh
